OpenWhisk/actions/nlcTrainer.js

```diff
@@ -0,0 +1,432 @@
+/*
+ * Licensed Materials - Property of IBM
+ * (C) Copyright IBM Corp. 2016. All Rights Reserved.
+ * US Government Users Restricted Rights - Use, duplication or
+ * disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+ */
+/*eslint strict: ["off", "global"]*/
+function main(params) {
+	'use strict';  // OpenWhisk can't resolve main function if strict mode is enabled at global scope.
+	/**
+	 *  Action to train NLC instance using image metadata that BluePics stores in cloudant.
+	 *
+	 *  @param {string} cloudantUsername (required) - cloudant user used for BluePic
+	 *  @param {string} cloudantPassword (required) - cloudant password used for BluePic
+	 *  @param {string} cloudantDbName (required) - cloudant database name used for BluePic
+	 *  @param {string} nlcUsername (required) - username for NLC
+	 *  @param {string} nlcPassword (required) - password for NLC
+	 *  @param {string} nlcUrl (required) - URL for NLC
+	 *  @param {string} nlcClassifier (optional) - Name used for classifier associated with this action.
+	 *  @param {string} logLevel (optional) - Set the log level used for this action.  Default is INFO.
+	 *  @param {string} nlcForceTraining (optional) - Set to "true" to skip training checks and force NLC training
+	 *  @param {string} NLC_LIMIT_NUM_CLASSES (optional) - Max # of class to train NLC with.
+	 *  @param {string} NLC_LIMIT_TEXT_LENGTH (optional) - Max length of training data statement.
+	 *  @param {string} NLC_LIMIT_MIN_RECORDS (optional) - Minimum # of training data records needed to train NLC.
+	 *  @param {string} NLC_LIMIT_MAX_RECORDS (optional) - Max # of records that can be used to train NLC.
+	 *
+	 */
+	const watson = require('watson-developer-cloud');
+	const Cloudant = require('cloudant');
+
+	var log4js = require('log4js');
```

Comment:
We have been using `winston` for logging in other libraries where we don't have Hubot's logger.  I don't have a preference either way, but it would be good to standardize on using the same library across all our projects.